src/tiled/objecttemplatemodel.cpp

```diff
@@ -153,6 +153,24 @@ bool ObjectTemplateModel::addNewDocument(TemplateGroupDocument *document)
     return true;
 }
 
+bool ObjectTemplateModel::addDocument(TemplateGroupDocument *document)
+{
+    if (!document)
```

Comment:
Okay, my understanding now is that we don't need to handle invalid scenarios that won't happen and the Q_Assert helps ensure that while coding.